### PR TITLE
feat: initial implementation for the Web Bluetooth spec

### PR DESCRIPTION
**Pull request by OrKoN**
This PR implements Web Bluetooth specification for WebDriver BiDi using CDP.

### CODE REVIEW

File: src/bidiMapper/CommandProcessor.ts [hunk 1]
```diff
@@ -60,6 +61,7 @@ type CommandProcessorEventsMap = {
 
 export class CommandProcessor extends EventEmitter<CommandProcessorEventsMap> {
   // keep-sorted start
+  #bluetoothProcessor: BluetoohProcessor;
```
Comment (mathiasbynens) [inline src/bidiMapper/CommandProcessor.ts:64]: minor typo:

```suggestion
  #bluetoothProcessor: BluetoothProcessor;
```

File: src/bidiMapper/CommandProcessor.ts [hunk 2]
```diff
@@ -31,6 +31,7 @@ import type {Result} from '../utils/result.js';
 
 import {BidiNoOpParser} from './BidiNoOpParser.js';
 import type {BidiCommandParameterParser} from './BidiParser.js';
+import {BluetoothProcessor} from './modules/bluetooth/BluetoothProcessor';
```
Comment (OrKoN) [inline src/bidiMapper/CommandProcessor.ts:34]:
```suggestion
import {BluetoothProcessor} from './modules/bluetooth/BluetoothProcessor.js';
```

**Comment by OrKoN** [comment]
@alexnj it looks like handleDevicePrompt should work as it only passed the ids down to CDP https://github.com/GoogleChromeLabs/chromium-bidi/pull/2060/files#diff-72654de89f8552b0cde4fea49eda0b5bd82dc3be3364aaa11dbbed03db83cc4dR52

File: src/bidiMapper/BidiServer.ts [hunk 3]
```diff
@@ -26,6 +26,7 @@ import type {Result} from '../utils/result.js';
 import type {BidiCommandParameterParser} from './BidiParser.js';
 import type {BidiTransport} from './BidiTransport.js';
 import {CommandProcessor, CommandProcessorEvents} from './CommandProcessor.js';
+import {BluetoothProcessor} from './modules/bluetooth/BluetoothProcessor';
```
Comment (OrKoN) [inline src/bidiMapper/BidiServer.ts:29]:
```suggestion
import {BluetoothProcessor} from './modules/bluetooth/BluetoothProcessor.js';
```

File: src/bidiMapper/CommandProcessor.ts [hunk 4]
```diff
@@ -31,6 +31,7 @@ import type {Result} from '../utils/result.js';
 import {BidiNoOpParser} from './BidiNoOpParser.js';
 import type {BidiCommandParameterParser} from './BidiParser.js';
 import type {MapperOptions} from './BidiServer.js';
+import type {BluetoothProcessor} from './modules/bluetooth/BluetoothProcessor';
```
Comment (OrKoN) [inline src/bidiMapper/CommandProcessor.ts:34]:
```suggestion
import type {BluetoothProcessor} from './modules/bluetooth/BluetoothProcessor.js';
```

File: src/protocol/chromium-bidi.ts [hunk 5]
```diff
@@ -104,7 +119,14 @@ export type CommandResponse =
   | WebDriverBidi.CommandResponse
   | Cdp.CommandResponse;
 
-export type Event = WebDriverBidi.Event | Cdp.Event;
+export type BluetoothEvent = {
+  type: 'event';
+} & (
+  | WebDriverBidiBluetooth.Bluetooth.RequestDevicePromptOpened
```
Comment (sadym-chromium) [inline src/protocol/chromium-bidi.ts:125]: Shouldn't `WebDriverBidiBluetooth.Bluetooth.RequestDevicePromptOpened` be `WebDriverBidi.Extensible` as well?

File: src/protocol/chromium-bidi.ts [hunk 6]
```diff
@@ -96,6 +106,11 @@ export type Command = (
       // not re-define it. Therefore, it's not part of generated types.
       id: WebDriverBidi.JsUint;
     } & WebDriverBidiPermissions.PermissionsCommand)
+  | ({
+      // id is defined by the main WebDriver BiDi spec and extension specs do
```
Comment (sadym-chromium) [inline src/protocol/chromium-bidi.ts:110]: Can we update the type generator, so that it generates the `id` for all the commands?

**Review by sadym-chromium — APPROVED** [review]

File: src/protocol/chromium-bidi.ts (hunk 6 above)
Comment (OrKoN) [reply src/protocol/chromium-bidi.ts:110]: Would you like to look into it? Note that this file is not generated.

File: src/protocol/chromium-bidi.ts (hunk 5 above)
Comment (OrKoN) [reply src/protocol/chromium-bidi.ts:125]: Not sure, but it does not seem to cause any issues currently.

**Pull request merged**